src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,94 @@
+package tasks;
+
+import parser.Parser;
+import exceptions.ArgumentException;
+
+import java.time.LocalDate;
+
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public static Task makeTask(String[] taskData) throws ArgumentException {
+        String taskType = taskData[0];
+        switch (taskType) {
+        case "T":
+            return new ToDo(taskData[2], taskData[1]);
+        case "D":
+            String[] args = Parser.parseDeadlineArgument(taskData[2]);
+            return new Deadline(args[0].trim(), LocalDate.parse(args[1].trim()), taskData[1]);
+        case "E":
+            args = Parser.parseEventArgument(taskData[2]);
+            return new Event(args[0].trim(), LocalDate.parse(args[1].trim()),
+                    LocalDate.parse(args[2].trim()), taskData[1]);
+        default:
+            throw new ArgumentException("Did not recognize task type " + taskType);
+        }
+    }
+
+    public static Task makeTask(String taskType, String taskArgument) throws ArgumentException {
+        switch (taskType) {
```

Comment:
I like your indentations here! Oftentimes people indent the case clauses so good job!